### ISSUE TITLE
After upgrade from 12.119.2 to 13.0.0 misskey no longer boots

### ISSUE THREAD

**Issue by Sakrecoer** [syuilo/Misskey#9663]
## 💡 Summary

After upgrade from 12.119.2 to 13.0.0 misskey no longer boots. Upgrade was problematic, apparently 2GB free ram wasn't enough. Freeing up RAM allowed me to proceed with upgrade. Once accomplished, the instance no longer boots exits with 1 and a generic error message: `ELIFECYCLE`

I have exhausted my ideas to figure out what is happening. But i suspect Misskey no longer supports postgres 12? It's either that or 2 CPU and 4GB ram is no longer sufficient, which strikes me as very odd.

Anyone got a clue how to further debug?
Many thanks in advance :raised_hands: 

```
INFO *  [core boot]     Welcome to Misskey!
INFO *  [core boot]     Misskey v13.0.0
INFO *  [core boot env] NODE_ENV: production
INFO *  [core boot nodejs]      Version v18.13.0 detected.
ERR  *  [core boot]     Fatal error occurred during initialization
INFO *  [core]  The process is going to exit with code 1
 ELIFECYCLE  Command failed with exit code 1.
```

## 🥰 Expected Behavior

run `NODE_ENV=production pnpm run start` misskey starts

## 🤬 Actual Behavior

run `NODE_ENV=production pnpm run start` misskey fails

## 📌 Environment

Misskey version: 13.0.0
Your OS: Ubuntu 20.04
Postgres: 12
Nodejs: v18.13.0 
2 CPU
4GB ram

**Comment by CyberRex0** [comment]
Yes, Postgres 12 is no longer supported.
You should upgrade Postgres to 15.

**Comment by syuilo** [comment]
ただ現状ポスグレ12でも何も問題はないから謎

**Comment by Sakrecoer** [comment]
Thank you very much for your prompt replies, @syuilo and @CyberRex0. If postgres 12 is still supported, any idea what i can do to investigate further? I have been unsuccessful at getting `pnpm` to be more verbose, and i can't seem to find any log files.

**Comment by aengl** [comment]
Since the Node.js version was printed, the error message suggests that the exception was thrown in [function loadConfigBoot()](https://github.com/misskey-dev/misskey/blob/206718032813c6dfde93e545d5b1e134784d2aa7/packages/backend/src/boot/master.ts#L115).

Which means that there's an error in the configuration file.

I think it would be very helpful if that function would log the exception instead of just re-raising it, something like `configLogger.error(exception.toString(), null, true)`. Otherwise it's going to be very difficult to debug configuration errors.

`connectDb()` is commented out for the initialization, so it's very unlikely that this has anything to do with your postgres version.

**Comment by Sakrecoer** [comment]
Thanks a million @aengl !! 

Very strange, i did not alter the config file :thinking: But maybe some setting is no longer valid? I did set `signToActivityPubGet: true` after issue federating with a couple of Mastodon instances and Peertube.

Edit: commenting `signToActivityPubGet: true` out didn't cut it.

**Comment by Sakrecoer** [comment]
This is how the issue was resolved:

On the recommendation of @aengl, i manually edited `packages/backend/src/boot/master.ts` below line 69, i add a new line:

```
bootLogger.error(`Error was: ${e.toString()}`, null, true);
```

I then cleaned up with `pnpm run clean` and `pnpm run clean-all`, went through all the steps of upgrade.

At boot, misskey now complained about the following:

`ERR  *  [core boot]     Error was: Error: ENOENT: no such file or directory, open '/home/misskey/misskey/packages/backend/built/boot/../../../../.node-version'`

Low and behold for some obscure reason i had deleted `.node-version`, so i added it back, and now the instance is back up and running!

Big thank you @aengl :purple_heart: :yellow_heart: 

Closing the issue.
